smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/SwiftDependency.kt

```diff
@@ -11,189 +11,241 @@ class SwiftDependency(
     override val target: String,
     private val branch: String? = null,
     private val version: String,
-    private val url: String,
+    private val location: String,
     private val localPath: String,
     override var packageName: String,
+    private val distributionMethod: DistributionMethod,
 ) : Dependency {
 
+    enum class DistributionMethod {
+        SPR, GIT
+    }
+
     companion object {
-        val NONE = SwiftDependency("", "", "", "", "", "")
-        val XCTest = SwiftDependency("XCTest", null, "", "", "", "")
+        val NONE = SwiftDependency(
+            "",
+            "",
+            "",
+            "",
+            "",
+            "",
+            DistributionMethod.GIT,
+        )
+        val SWIFT = SwiftDependency(
+            "Swift",
+            "",
+            "",
+            "",
+            "",
+            "",
+            SwiftDependency.DistributionMethod.GIT,
+        )
+        val XCTest = SwiftDependency(
+            "XCTest",
+            null,
+            "",
+            "",
+            "",
+            "",
+            DistributionMethod.GIT,
+        )
         val CRT = SwiftDependency(
             "AwsCommonRuntimeKit",
             null,
             "0.30.0",
             "https://github.com/awslabs/aws-crt-swift",
             "",
             "aws-crt-swift",
+            DistributionMethod.GIT,
         )
         val CLIENT_RUNTIME = SwiftDependency(
```

Comment:
This & the other SwiftDependencies below are updated for hosting on the package registry.